src/PSR7/Validators/CookiesValidator/ServerRequestCookieValidator.php

```diff
@@ -37,38 +37,14 @@ public function __construct(array $specs)
     public function validate(OperationAddress $addr, MessageInterface $message) : void
     {
         Assert::isInstanceOf($message, ServerRequestInterface::class);
-        $this->checkRequiredCookies($addr, $message);
-        $this->checkCookiesAgainstSchema($addr, $message);
-    }
-
-    /**
-     * @throws InvalidCookies
-     */
-    private function checkRequiredCookies(OperationAddress $addr, ServerRequestInterface $message) : void
-    {
-        foreach ($this->specs as $cookieName => $spec) {
-            if ($spec->required && ! array_key_exists($cookieName, $message->getCookieParams())) {
-                throw InvalidCookies::becauseOfMissingRequiredCookie($cookieName, $addr);
-            }
-        }
-    }
-
-    /**
-     * @throws InvalidCookies
-     */
-    private function checkCookiesAgainstSchema(OperationAddress $addr, ServerRequestInterface $message) : void
-    {
-        foreach ($message->getCookieParams() as $cookieName => $cookieValue) {
-            if (! isset($this->specs[$cookieName])) {
-                continue;
-            }
-
-            $validator = new SchemaValidator($this->detectValidationStrategy($message));
-            try {
-                $validator->validate($cookieValue, $this->specs[$cookieName]->schema);
-            } catch (SchemaMismatch $e) {
-                throw InvalidCookies::becauseValueDoesNotMatchSchema($cookieName, $cookieValue, $addr, $e);
-            }
+        $validator = new ArrayValidator($this->specs);
+
+        try {
+            $validator->validateArray($addr, $message->getCookieParams(), $this->detectValidationStrategy($message));
+        } catch (RequiredParameterMissing $e) {
+            throw InvalidCookies::becauseOfMissingRequiredCookie($e->name(), $addr);
```

Comment:
Actually, as far as I can see, you do the rethrow I mentioned here, so there is really no need to bound `ArrayValidtor` to PSR-7 stuff (`$addr`)